subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/vcs/VcsDependencyResolver.java

```diff
@@ -108,7 +108,7 @@ private File getBaseWorkingDir() {
         ProjectInternal rootProject = projectRegistry.getRootProject();
         File baseWorkingDir = null;
         if (rootProject!=null) {
-            baseWorkingDir = new File(rootProject.getBuildDir(), "vcsWorkingDirs");
+            baseWorkingDir = new File(rootProject.getRootDir(), ".gradle/vcsWorkingDirs");
```

Comment:
This isn't going to be the right thing always.  e.g., people could do this (but probably rarely do):  `gradle --project-cache-dir myprojcache build`

Then the directory is `myprojectcache` and not `.gradle`. 

We need to do something different for this for https://github.com/gradle/gradle-native/issues/188.